File: WhatsAppAutoReply/app/src/main/java/com/whatsappauto/reply/receivers/BootReceiver.java
package com.whatsappauto.reply.receivers;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.whatsappauto.reply.services.MessageSchedulerService;

public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            // Start the message scheduler service
            Intent serviceIntent = new Intent(context, MessageSchedulerService.class);
            context.startService(serviceIntent);
        }
    }
}

File: app/src/main/java/com/whatsapp/auto/responder/receiver/BootReceiver.kt
package com.whatsapp.auto.responder.receiver

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.util.Log
import com.whatsapp.auto.responder.worker.ScheduledMessageWorker

class BootReceiver : BroadcastReceiver() {

    companion object {
        private const val TAG = "BootReceiver"
    }

    override fun onReceive(context: Context, intent: Intent) {
        when (intent.action) {
            Intent.ACTION_BOOT_COMPLETED,
            Intent.ACTION_MY_PACKAGE_REPLACED -> {
                Log.d(TAG, "Boot completed or package replaced")
                
                // Start the scheduled message worker
                ScheduledMessageWorker.startPeriodicWork(context)
                
                Log.d(TAG, "Scheduled message worker started")
            }
        }
    }
}